webdriver/tests/close_window/user_prompts.py

```diff
@@ -24,23 +24,33 @@ def test_handle_prompt_ignore():
 def test_handle_prompt_accept(new_session, add_browser_capabilites):
     _, session = new_session({"capabilities": {
         "alwaysMatch": add_browser_capabilites({"unhandledPromptBehavior": "accept"})}})
-    session.window_handle = create_window(session)()
+    original_handle = session.window_handle
 
+    session.window_handle = create_window(session)()
     session.url = inline("<title>WD doc title</title>")
 
     create_dialog(session)("alert", text="dismiss #1", result_var="dismiss1")
     response = close(session)
     assert response.status == 200
+    session.window_handle = original_handle
     assert_dialog_handled(session, "dismiss #1")
```

Comment:
We want to make sure that the dialog on the newly opened window has been handled and not on the original window, right? In that case resetting the `window_handle` should come after the assertion.

Reply:
Hm, given that the dialog should have been accepted by the `close` command, the window is gone. Does it make sense to have the check at all for this command?

Reply:
I suppose it makes sense that we assert that no dialog is present, which is what assert_dialog_handled does. However, the call to Get Alert Text (which is what assert_dialog_handled uses to validate whether an alert is present) requires a valid, open top-level browsing context, so the reset of the current window handle must come first.

Reply:
So lets have a look at the spec for Switch To Window:

> If there is an active user prompt, that prevents the focussing of another top-level browsing context, return error with error code unexpected alert open. 

So similar to other commands switching to a different window will also accept the dialog in this test. So using `assert_dialog_handled` afterward would be a no-op. Best would still be to check that the newly created window is gone, and if not make the dialog check.